sdk/src/test/scala/com/horizen/account/wallet/AccountWalletTest.scala

```diff
@@ -195,4 +195,40 @@ class AccountWalletTest
     assertEquals("SidechainWallet different exception expected during removing new Secret.", expectedException, failureResult.failed.get)
   }
 
+  @Test
+  def testGenerateNextSecret(): Unit = {
+    val mockedSecretStorage: SidechainSecretStorage = mock[SidechainSecretStorage]
+
+    val accountWallet = new AccountWallet(
+      "seed".getBytes(),
+      mockedSecretStorage)
+
+    val storageList = ListBuffer[Secret]()
+    val secret1 = getPrivateKey25519("seed1".getBytes())
+    val secret2 = getPrivateKey25519("seed2".getBytes())
+    storageList += secret1
+    storageList += secret2
+    Mockito.when(mockedSecretStorage.getAll).thenReturn(storageList.toList)
+    Mockito.when(mockedSecretStorage.add(ArgumentMatchers.any[Secret])).thenReturn(Try{mockedSecretStorage})
+
+    // Prepare block ID and corresponding version
+    val blockId = new Array[Byte](32)
+    Random.nextBytes(blockId)
```

Comment:
not used, can be removed

Reply:
Removed

Reply:
@i-Alex  I removed the part that you highlighted. accountWallet is used in the code in 2 places

Reply:
why do you talk about `accountWallet` ?
comment is about `blockId` usage